gson/src/main/java/com/google/gson/JsonPrimitive.java

```diff
@@ -137,108 +145,102 @@ public boolean isString() {
     return value instanceof String;
   }
 
-  /**
-   * convenience method to get this element as a String.
-   *
-   * @return get this element as a String.
-   */
+  // Don't add Javadoc, inherit it from super implementation; no exceptions are thrown here
   @Override
   public String getAsString() {
-    if (isNumber()) {
+    if (value instanceof String) {
+      return (String) value;
+    } else if (isNumber()) {
       return getAsNumber().toString();
     } else if (isBoolean()) {
       return ((Boolean) value).toString();
-    } else {
-      return (String) value;
     }
+    throw new AssertionError("Unexpected value type: " + value.getClass());
   }
 
   /**
-   * convenience method to get this element as a primitive double.
-   *
-   * @return get this element as a primitive double.
-   * @throws NumberFormatException if the value contained is not a valid double.
+   * @throws NumberFormatException {@inheritDoc}
```

Comment:
I don't think you need this. Just leaving out the doc comment altogether should have the same effect.

Reply:
It looks like javadoc does not inherit the documented exceptions in that case. So either there has to be a `@throws`, or I add a `throws NumberFormatException` to the method signature. What do you prefer?

Reply:
Oh, I think I've seen this before. The specified behaviour is that exception comments are only copied for exceptions that are declared in the `throws` clause. So what you have here is right. Sorry for the distraction.